command/meta.go

```diff
@@ -212,11 +226,96 @@ func (m *ApiMeta) initializeX509Config(client http.Client, clientCA []byte, cert
 	client.Transport.(*http.Transport).TLSClientConfig.MinVersion = tls.VersionTLS12
 	client.Transport.(*http.Transport).TLSClientConfig.PreferServerCipherSuites = true
 	client.Transport.(*http.Transport).TLSClientConfig.Certificates = []tls.Certificate{cert}
-	client.Transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify = true // TODO(jacobkiefer): Add a flag this.
+	client.Transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify = true // TODO(jacobkiefer): Add a flag for this.
 
 	return &client
 }
 
+func (m *ApiMeta) Authenticate() error {
+	auth := m.Config.Auth
+	if auth != nil && auth.Enabled && auth.OAuth2 != nil {
+		OAuth2 := auth.OAuth2
+		if !OAuth2.IsValid() {
+			return errors.New("Incorrect OAuth2 auth configuration.")
+		}
+
+		usr, err := user.Current()
+		if err != nil {
+			m.Ui.Error("Could not read current user from environment, failing.")
+			return err
+		}
+		credentialsLocation := filepath.Join(usr.HomeDir, ".spin", "credentials")
+		config := &oauth2.Config{
+			ClientID:     OAuth2.ClientId,
+			ClientSecret: OAuth2.ClientSecret,
+			RedirectURL:  "http://localhost:8085",
```

Comment:
What does hardcoding this do?

Reply:
Hardcodes the address of the local http server we use to catch the `code` response.  This is necessary because you have to whitelist redirect URLs when creating an OAuth2 client in your provider: https://www.spinnaker.io/setup/security/authentication/oauth/providers/google/.

Reply:
This is also consistent with what `gcloud` does for instance.